code/go/0chain.net/smartcontract/zcnsc/sc.go

```diff
@@ -25,35 +25,72 @@ const (
 	UpdateAuthorizerConfigFunc = "update-authorizer-config"
 	MintFunc                   = "mint"
 	BurnFunc                   = "burn"
+	AddToDelegatePoolFunc      = "addToDelegatePool"
+	DeleteFromDelegatePoolFunc = "deleteFromDelegatePool"
+	DistributeRewardsFunc      = "distributeRewards"
 )
 
 // ZCNSmartContract ...
 type ZCNSmartContract struct {
 	*smartcontractinterface.SmartContract
+	smartContractFunctions map[string]smartContractFunction
 }
 
 func NewZCNSmartContract() smartcontractinterface.SmartContractInterface {
-	var zcnCopy = &ZCNSmartContract{
+	var contract = &ZCNSmartContract{
 		smartcontractinterface.NewSC(ADDRESS),
+		make(map[string]smartContractFunction),
 	}
-	zcnCopy.setSC(zcnCopy.SmartContract, &smartcontract.BCContext{})
-	return zcnCopy
+
+	contract.InitSC()
+	contract.setSC(contract.SmartContract, &smartcontract.BCContext{})
+	return contract
 }
 
 // InitSC ...
-func (zcn *ZCNSmartContract) InitSC() {}
+func (zcn *ZCNSmartContract) InitSC() {
+	zcn.smartContractFunctions[MintFunc] = zcn.Mint
+	zcn.smartContractFunctions[BurnFunc] = zcn.Burn
+	zcn.smartContractFunctions[AddAuthorizerFunc] = zcn.AddAuthorizer
+	zcn.smartContractFunctions[DeleteAuthorizerFunc] = zcn.DeleteAuthorizer
+	zcn.smartContractFunctions[UpdateGlobalConfigFunc] = zcn.UpdateGlobalConfig
+	zcn.smartContractFunctions[UpdateAuthorizerConfigFunc] = zcn.UpdateAuthorizerConfig
+	zcn.smartContractFunctions[DistributeRewardsFunc] = zcn.DistributeRewards
```

Comment:
This is not a smart contract endpoint. Rather we should call the stake pool's DistributeRewards method whenever we pay rewards to stakeholders.

Instead, we would want a `collect_rewards` endpoint that will transfer the rewards previously distributed to delegate wallets. Similar to storages [collect_reward](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/collect_reward.go#L13).

Reply:
Done

Reply:
Done